component/src/main/java/org/wso2/extension/siddhi/io/cdc/util/CDCSourceUtil.java

```diff
@@ -0,0 +1,237 @@
+
+        String host;
+        int port;
+        String database;
+
+        //Add schema specific details to configMap
+
+        String[] splittedURL = url.split(":");
+        if (!splittedURL[0].equalsIgnoreCase("jdbc")) {
+            throw new WrongConfigurationException("Invalid JDBC url: " + url);
+        } else {
+            switch (splittedURL[1]) {
+                case "mysql": {
+
+                    //Extract url details
+                    String regex = "jdbc:mysql://(\\w*|[0-9]{1,3}\\.[0-9]{1,3}\\.[0-9]{1,3}\\.[0-9]{1,3}):" +
+                            "(\\d++)/(\\w*)";
+                    Pattern p = Pattern.compile(regex);
+                    Matcher matcher = p.matcher(url);
+                    if (matcher.find()) {
+                        host = matcher.group(1);
+                        port = Integer.parseInt(matcher.group(2));
+                        database = matcher.group(3);
+
+                    } else {
+                        throw new WrongConfigurationException("Invalid JDBC url: " + url);
+                    }
+
+                    //Add extracted url details to configMap.
+                    configMap.put(CDCSourceConstants.DATABASE_PORT, port);
+                    configMap.put(CDCSourceConstants.TABLE_WHITELIST, database + "." + tableName);
+                    configMap.put(CDCSourceConstants.DATABASE_HOSTNAME, host);
+
+                    //Add other MySQL specific details to configMap.
+                    configMap.put(CDCSourceConstants.CONNECTOR_CLASS, CDCSourceConstants.MYSQL_CONNECTOR_CLASS);
+
+                    break;
+                }
+                default: {
+                    throw new WrongConfigurationException("Unsupported schema. Expected schema: mysql, Found: "
+                            + splittedURL[1]);
+                }
+
+            }
+
+            //Add general config details to configMap
+            configMap.put(CDCSourceConstants.DATABASE_USER, username);
+            configMap.put(CDCSourceConstants.DATABASE_PASSWORD, password);
+
+            if (serverID == -1) {
+                Random random = new Random();
+                configMap.put(CDCSourceConstants.SERVER_ID, random.nextInt(1001) + 5400);
+            } else {
+                configMap.put(CDCSourceConstants.SERVER_ID, serverID);
+            }
+
+            //set the database server name if specified, otherwise set <host>_<port> as default
+            if (serverName.equals("")) {
+                configMap.put(CDCSourceConstants.DATABASE_SERVER_NAME, host + "_" + port);
+            } else {
+                configMap.put(CDCSourceConstants.DATABASE_SERVER_NAME, serverName);
+            }
+
+            configMap.put(CDCSourceConstants.OFFSET_STORAGE, InMemoryOffsetBackingStore.class.getName());
+            configMap.put(CDCSourceConstants.CDC_SOURCE_OBJECT, cdcSourceHashCode);
+
+            //set history file path.
+            configMap.put(CDCSourceConstants.DATABASE_HISTORY,
```

Comment:
I think we can get these lines to a single line.

Reply:
Fixed with commit: `Fix some suggested issues in PR#1.` 8a89f22